src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +16,36 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(actor);
+            transaction.commit();
+        } catch (HibernateException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add an actor: " + actor + " to DB.", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return actor;
     }
 
     @Override
     public Optional<Actor> get(Long id) {
-        return null;
+        Actor actor = null;
+        try (Session session = factory.openSession()) {
+            actor = session.get(Actor.class, id);
+        } catch (HibernateException e) {
+            throw new DataProcessingException("Can't get an actor with id: " + id
+                                               + " from DB.", e);
+        }
+        return Optional.ofNullable(actor);
```

Comment:
you can move it in try block and make it inline and remove actor instance
```suggestion
         try (Session session = factory.openSession()) {
            return Optional.ofNullable(session.get(Actor.class, id));
        } catch (HibernateException e) {
            throw new DataProcessingException("Can't get an actor with id: " + id
                                               + " from DB.", e);
        }
```